harness/context_manager.py

```diff
@@ -47,6 +47,9 @@ def __init__(
     def __call__(self, cmd, raise_error=True, **kwargs):
         try:
             combined_args = {**self.subprocess_args, **kwargs}
+            if "source" in cmd:
+                combined_args["shell"] = True
+                combined_args["executable"] = "/bin/bash"
```

Comment:
had issues with this on linux

Reply:
We removed `source` in favor of `.` as of this message. We also don't default the `shell` setting to True to mitigate the portability issues, overhead, and minor security risks that are associated with executing thru the system shell.

There are still a couple places we use `shell=True` (primarily in some installation commands), but in general we set it to `False` by default.